Cosminis/DataAccess/FriendsRepo.cs

```diff
@@ -0,0 +1,457 @@
+using DataAccess;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+
+namespace DataAccess;
+ 
+public class FriendsRepo : IFriendsDAO
+{
+    private readonly wearelosingsteamContext _context;
+    private readonly IUserDAO _userRepo;
+
+    public FriendsRepo(wearelosingsteamContext context, IUserDAO userRepo)
+    {
+        _context = context;
+        _userRepo = userRepo;
+    }
+
+    public List<Friends> GetAllRelationships()
+    {
+        return _context.Friends.ToList();
+    }
+
+    public List<Friends> ViewAllFriends(int userIdToLookup)
+    {
+        User userToLookup = _context.Users.Find(userIdToLookup);
+
+        IEnumerable<Friends> friendsQuery =
+            (from Friends in _context.Friends
+            where (Friends.UserIdTo == userToLookup.UserId) || (Friends.UserIdFrom == userToLookup.UserId)
+            select Friends);
+
+        if(friendsQuery == null)
+        {
+            throw new ResourceNotFound("This is one lonely user. At this point they should simply quit social media.");
+        }
+
+        try
+        {       
+            List<Friends> friendsList = friendsQuery.ToList();
+            
+            if(friendsList.Count() < 1)
+            {
+                throw new ResourceNotFound();
+            }
+
+            return friendsList;
+        }
+        catch(Exception E)
+        {
+            throw;
+        }                                  
+    }
+    
+    public Friends FriendsByUserIds(int searchingUserId, int user2BeSearchedFor)
+    {
+        User searchingUser = _context.Users.Find(searchingUserId);
+        User friend2BFound = _context.Users.Find(user2BeSearchedFor);
+
+        List<Friends> quieriedFriendsList = ViewAllFriends((int)searchingUser.UserId);
+
+        List<Friends> searchedFriendsList = ViewAllFriends((int)friend2BFound.UserId);
+
+        Friends returnRelationship = new Friends();
+
+        foreach(Friends friendInstance in quieriedFriendsList)
+        {
+            if(searchedFriendsList.Contains(friendInstance))
+            {
+                returnRelationship = friendInstance;
+            }
+        }
+        return returnRelationship;
+    }
+    
+    public Friends EditFriendship(int editingUserID, int user2BeEdited, string status)//this does not delete removed/blocked friends. I may add this later.
+    {   
+        User editingUser = _context.Users.Find(editingUserID);
+        User friend2BeEdited = _context.Users.Find(user2BeEdited);
+        
+        Friends statusToBeEdited = FriendsByUserIds((int)editingUser.UserId, (int)friend2BeEdited.UserId);
+
+        statusToBeEdited.Status = status;
+
+        _context.SaveChanges();
+
+        _context.ChangeTracker.Clear(); 
+
+        return statusToBeEdited;  
+    }
+
+    public Friends SearchByRelationshipId(int relationshipId)
+    {
+        return _context.Friends.FirstOrDefault(friends => friends.RelationshipId == relationshipId) ?? throw new ResourceNotFound();
```

Comment:
_context.Friends.Find() achieves a very similar functionality, but this isn't hurting anyone

Reply:
GTK